data/battle_anim_scripts.s

```diff
@@ -2549,36 +2549,36 @@ Move_THRASH:
 Move_SING:
 	loadspritegfx ANIM_TAG_MUSIC_NOTES
 	monbg ANIM_DEF_PARTNER
-	createvisualtask sub_8102BE8, 2
+	createvisualtask AnimTask_MusicNotesRainbowBlend, 2
 	waitforvisualfinish
 	panse_1B SE_W047, SOUND_PAN_ATTACKER, SOUND_PAN_TARGET, +2, 0
-	createsprite gUnknown_08593008, ANIM_TARGET, 2, 7, 0, 12
+	createsprite gWavingMusicNotesSpriteTemplate, ANIM_TARGET, 2, 7, 0, 12
 	delay 5
-	createsprite gUnknown_08593008, ANIM_TARGET, 2, 6, 1, 12
+	createsprite gWavingMusicNotesSpriteTemplate, ANIM_TARGET, 2, 6, 1, 12
 	delay 5
-	createsprite gUnknown_08593008, ANIM_TARGET, 2, 1, 2, 12
+	createsprite gWavingMusicNotesSpriteTemplate, ANIM_TARGET, 2, 1, 2, 12
 	delay 5
-	createsprite gUnknown_08593008, ANIM_TARGET, 2, 2, 3, 12
+	createsprite gWavingMusicNotesSpriteTemplate, ANIM_TARGET, 2, 2, 3, 12
 	delay 5
-	createsprite gUnknown_08593008, ANIM_TARGET, 2, 3, 0, 12
+	createsprite gWavingMusicNotesSpriteTemplate, ANIM_TARGET, 2, 3, 0, 12
```

Comment:
Is this supposed to be plural?  Also, I would suggest `Wavy` instead of `Waving`.  `gWavyMusicNoteSpriteTemplate`